### PR TITLE
FIX: Address corner cases and high-load tuning with acctst

### LINKED ISSUE #323: Pass acctst
A tracking issue for any problems we encounter while running the caproto servers against the regression tester in epics-base, `acctst`.

### PR DESCRIPTION
**Pull request by danielballan**
See #323 for progress updates.

### CODE REVIEW

File: caproto/server/common.py [hunk 1]
```diff
@@ -314,7 +316,18 @@ def __init__(self, circuit, client, context):
                         "High load. Batched %d commands (%dB) with %.4fs latency.",
                         len_commands, commands_bytes,
                         now - deadline + HIGH_LOAD_TIMEOUT)
-                await self.send(*commands)
+
+                # Ensure at the last possible moment that we don't send
+                # respones for Subscriptions that have been canceled at some
```
Comment (klauer) [inline caproto/server/common.py]: `responses`

File: caproto/server/common.py [hunk 2]
```diff
@@ -464,9 +478,8 @@ def get_db_entry():
                         status=write_status,
                         data_count=db_entry.length
                     )
-
-                if client_waiting:
-                    await self.send(response_command)
+                    if client_waiting:
```
Comment (tacaswell) [inline caproto/server/common.py]: might as well move this up to include creating the response?  Do we have to turn the crank an creating that even if the client is not waiting for it back?

Comment (tacaswell) [reply caproto/server/common.py]: This is the first time in python that I have missed the extra scoping you can get with `{}` in c++...

File: caproto/server/common.py [hunk 3]
```diff
@@ -704,29 +704,21 @@ def __getitem__(self, pvname):
...
-                # A specific sub_id has been specified, which means this update
-                # is due to a *new* Subscription, and should only go to that
-                # specific Subscription. Other Subscriptions have already been
-                # sent this update and should not receive it again. Find which
-                # Subscription that is....
+                # A specific Subscription has been specified, which means this
+                # specific update was prompted by Subscription being new, not
+                # prompted by a new value. The update should only be sent to
+                # that specific Subscription.
+                subs = (sub,)
```
Comment (tacaswell) [inline caproto/server/common.py]: super nit-pick, make this a list to match the type used above?  Or replace the above with a tuple comprehension?

Comment (danielballan) [reply caproto/server/common.py]: Reasonable.

**Pull request merged**